src/db.c

```diff
@@ -860,6 +888,46 @@ int parseScanCursorOrReply(client *c, robj *o, unsigned long *cursor) {
     return C_OK;
 }
 
+char *obj_type_name[OBJ_TYPE_MAX] = {
+    "string", 
+    "list", 
+    "set", 
+    "zset", 
+    "hash", 
+    "", /* module type is special */
+    "stream"
+};
+
+long long getObjectTypeByName(char *name) {
+    for (long long i = 0; i < OBJ_TYPE_MAX; i++) {
+        if (!strcasecmp(name, obj_type_name[i])) {
+            return i;
+        }
+    }
+
+    moduleType *mt = moduleTypeLookupModuleByName(name);
+    if (mt != NULL) return -(REDISMODULE_TYPE_SIGN(mt->id));
+
+    return LLONG_MAX;
+}
+
+char *getObjectTypeName(robj *o) {
+    if (o == NULL) {
+        return "none";
+    }
+
+    if (o->type == OBJ_MODULE) {
+        moduleValue *mv = o->ptr;
+        return mv->type->name;
+    }
+
+    if (o->type < 0 || o->type > OBJ_TYPE_MAX) {
```

Comment:
```suggestion
    if (o->type < 0 || o->type >= OBJ_TYPE_MAX) {
```
on the other hand, `default: type = "unknown"; break;` is dead code in the origin code, maybe we can change it to `assert` instead.

Reply:
Agree, it does look like dead code.